core/src/main/scala/org/bykn/bosatsu/LetFree.scala

```diff
@@ -2,7 +2,6 @@ package org.bykn.bosatsu
 
 import cats.data.{NonEmptyList, State}
 import cats.implicits._
-import cats.Id
 import rankn._
 
```

Comment:
Can we add a comment to the class and explain what this is for (and AST without Lets I think), and why we would want it.

For instance, the motivation of the Matchless AST is that pattern matching is rather complex and a sublanguage beyond the basic lambda calculus. Why do we want to remove lets? I think it is part of a plan for normalizing expressions.

But since we still have patterns here, we should comment on pattern normalization, and how that plays a role.